src/message/messageActionSheet.js

```diff
@@ -173,7 +173,10 @@ const deleteTopic = async ({ auth, stream, topic, dispatch, _ }) => {
       );
     });
   if (await AsyncAlert()) {
-    await dispatch(deleteMessagesForTopic(stream, topic));
+    const sub = subscriptions.find(x => x.name === stream);
+    if (sub) {
+      await dispatch(deleteMessagesForTopic(sub.stream_id, topic));
```

Comment:
Is the added `if (sub)` conditional a functional change? If so, let's be sure it's a change that we want, describe it in the commit message, and un-mark the commit as NFC.

If it is NFC, I'm not seeing that that's obvious; that should get an explanation if it turns out to be the case, I think. 🙂

Reply:
Ah, looking back after posting the review: if we really expect `sub` to exist every single time, this is probably another good place to use `invariant` (see my other comments where I mention that).

Reply:
Using invariant in the new revision,

I think `subscriptions` will not change at this position i.e. it will keep using the same value that `showTopicActionSheet` was called with. even if it is updated by react in between the execution of this function.
I am not sure if that is intentional though (since such a change can happen when the user is about to interact with the action sheet.)

So if `showTopicActionSheet` is called with right stream name and Subscription object here,  `(sub !== undefined)` will always be true.
Hence I am using invariant.

**Why i think `subscriptions` will not change at this position?**
I think this is an example of "stale closure" (more info at: https://reactjs.org/docs/hooks-faq.html#why-am-i-seeing-stale-props-or-state-inside-my-function)

Essentially what I could infer from it was that when we pass a state to a function and the state updates, our function creates a closure with the value of state that was passed to it when the function was called and it keeps using that outdated data even when the state updates.
Just in case this is unnecessary a way to solve this is by using `useRef` (for functional components) and its equivalent for class. Here is a blog post discussing this: https://css-tricks.com/dealing-with-stale-props-and-states-in-reacts-functional-components/